code/datums/components/attachment.dm

```diff
@@ -0,0 +1,165 @@
+#define COMSIG_ATTACHMENT_ATTACH "attach-attach"
+#define COMSIG_ATTACHMENT_DETACH "attach-detach"
+#define COMSIG_ATTACHMENT_EXAMINE "attach-examine"
+#define COMSIG_ATTACHMENT_EXAMINE_MORE "attach-examine-more"
+#define COMSIG_ATTACHMENT_PRE_ATTACK "attach-pre-attack"
+#define COMSIG_ATTACHMENT_ATTACK "attach-attacked"
+#define COMSIG_ATTACHMENT_UPDATE_OVERLAY "attach-overlay"
+
+//Basically the parts of guns where shit attaches to.
+#define COMSIG_ATTACHMENT_GET_SLOT "attach-slot-who"
+	#define ATTACHMENT_SLOT_MUZZLE "attach-slot-muzzle"
```

Comment:
Why are all these tabbed in once?

Reply:
Probably error on my part when bringing the code over. Didn't seem to cause an issue (?)